### PR TITLE
Add virhunter tool

### PR DESCRIPTION
**Pull request by bdartigues**
Fixes https://github.com/galaxyproject/tools-iuc/issues/4778

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/virhunter/macros.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<macros>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="1.16">numpy</requirement>
+            <requirement type="package" version="1.78">biopython</requirement>
+            <requirement type="package" version="1.2.1">pandas</requirement>
+            <requirement type="package" version="1.0.0">joblib</requirement>
+            <requirement type="package" version="0.23.2">scikit-learn</requirement>
```
Comment (bgruening) [inline tools/virhunter/macros.xml:8]: Can you try adding here the tensorflow dependency and remove the pip install from the tool. Would be interesting to see if the CI is happy.

Comment (bdartigues) [reply tools/virhunter/macros.xml:8]: Done

Comment (bdartigues) [reply tools/virhunter/macros.xml]: it seems that i found the right conf with tf=2.9.1 but i need to change the test data accordingly

Comment (bdartigues) [reply tools/virhunter/macros.xml]: Dear All, are there any news about the approving review of the virhunter tool. For now, we used to test the tool, some fake model data but how can we incorporate our models without github filesize restriction ? Thanks in advance

**Comment by gallardoalba** [comment]
I think it would be nice to include an additional functionality to this tool in order to allow users to generate their training models in Galaxy https://github.com/cbib/virhunter#training-your-own-model

File: tools/virhunter/virhunter.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+
+    mkdir -p '${predicted_fragments.extra_files_path}' &&
+    python '$__tool_directory__/predict.py'
+        --test_ds '${fasta_file}'
+        --weights '${weights.fields.path}'
+        --out_path '${predicted_fragments.extra_files_path}'
+        --return_viral True
+        --limit 750
```
[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Hi Cristóbal Gallardo, Thanks for all theses suggestions. I commited a updated version

[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Done

[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Done

File: tools/virhunter/virhunter.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        <param name="weights" type="select" label="Select a reference model" help="If your model of interest is not listed, contact the Galaxy team">
+            <options from_data_table="virhunter_models">
+                <validator type="no_options" message="No models are available for the selected input dataset" />
+            </options>
+        </param>
+    </inputs>
+    <outputs>
+        <data format="csv" name="predicted_fragments" from_work_dir="predicted_fragments.csv" label="${tool.name} on ${on_string}: predicted fragments"/>
+        <data format="csv" name="predicted" from_work_dir="predicted.csv" label="${tool.name} on ${on_string}: predicted "/>
+        <data format="fasta" name="viral" from_work_dir="viral.fasta" label="${tool.name} on ${on_string}: viral fasta file" />
```
[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Done

File: tools/virhunter/virhunter.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<tool id="virhunter" name="virhunter" version='1.0.0' >
+    <description>
+        VirHunter is a deep learning method to identify viruses in sequencing datasets..
```
[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Done

File: tools/virhunter/macros.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<macros>
+    <xml name="requirements">
```
[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/macros.xml]: Done

File: tools/virhunter/virhunter.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    </description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
```
[the remark this answers is not in the document]
Comment (bdartigues) [reply tools/virhunter/virhunter.xml]: Done

**Comment by gallardoalba** [comment]
Thank you so much for the contribution @bdartigues, it seems very interesting tool.

**Comment by bdartigues** [comment]
> Thank you so much for the contribution @bdartigues, it seems a very interesting tool.

So what is the next step for me to incorporate my real models files given that githhub limit is 2Mo ? Bjoern gruenind told me to go directly through the administrators so that they install the model files. In total, these files (8 models) represent only 53 MB in all. Many thanks in advance. Benjamin

**Comment by gallardoalba** [comment]
> > Thank you so much for the contribution @bdartigues, it seems a very interesting tool.
> 
> So what is the next step for me to incorporate my real models files given that githhub limit is 2Mo ? Bjoern gruenind told me to go directly through the administrators so that they install the model files. In total, these files (8 models) represent only 53 MB in all. Many thanks in advance. Benjamin

Probably models can be installed manually at the moment until a better solution is found @bgruening

**Comment by gallardoalba** [comment]
> > > > Thank you so much for the contribution @bdartigues, it seems a very interesting tool.
> > > 
> > > 
> > > So what is the next step for me to incorporate my real models files given that githhub limit is 2Mo ? Bjoern gruenind told me to go directly through the administrators so that they install the model files. In total, these files (8 models) represent only 53 MB in all. Many thanks in advance. Benjamin
> > 
> > 
> > Probably models can be installed manually at the moment until a better solution is found @bgruening
> 
> I have a small .sh script to install models on the weights folder. Do you want me to add in ? or maybe i can sen you a mail with the script attached ?

Thank you so much, could you share it by using gist.github.com?

**Comment by bdartigues** [comment]
> > > > > Thank you so much for the contribution @bdartigues, it seems a very interesting tool.
> > > > 
> > > > 
> > > > So what is the next step for me to incorporate my real models files given that githhub limit is 2Mo ? Bjoern gruenind told me to go directly through the administrators so that they install the model files. In total, these files (8 models) represent only 53 MB in all. Many thanks in advance. Benjamin
> > > 
> > > 
> > > Probably models can be installed manually at the moment until a better solution is found @bgruening
> > 
> > 
> > I have a small .sh script to install models on the weights folder. Do you want me to add in ? or maybe i can sen you a mail with the script attached ?
> 
> Thank you so much, could you share it by using [gist.github.com](gist.github.com)?

Yes ! here is the link to the script https://gist.github.com/bdartigues/aa35064ea7fbd47fa20c49d8ae64c42a

**Review by gallardoalba — APPROVED** [review]

**Comment by bdartigues** [comment]
Great news that you've merged the tool but I thought you'd integrate the model in the tool-data/weight and test-data/weight folder and I might modify some  lines that i'd commented for testing purposes before the merge. In the state, the tool is not really functional since it uses a synthetic model (test model) produced for the occasion. How can i proceed to commit again ?

**Comment by gallardoalba** [comment]
> Great news that you've merged the tool but I thought you'd integrate the model in the tool-data/weight and test-data/weight folder and I might modify some lines that i'd commented for testing purposes before the merge. In the state, the tool is not really functional since it uses a synthetic model (test model) produced for the occasion. How can i proceed to commit again ?

Could you open a PR against the IUC repository?

**Pull request merged**